### PR TITLE
check whether the user has a google account

### PR DESCRIPTION
**Pull request by yuhuyoyo**
only run the custom proxy agent when the user has non-google-account=true

https://verily.atlassian.net/browse/BENCH-2608

### CODE REVIEW

File: startupscript/vertex-ai-user-managed-notebook/post-startup.sh [hunk 1]
```diff
@@ -830,18 +830,24 @@ fi
...
-readonly TERRA_GCP_NOTEBOOK_RESOURCE_NAME
-if [[ -n "${APP_PROXY}" ]]; then
-  emit "Using custom Proxy Agent"
-  RESOURCE_ID=$(get_metadata_value "instance/attributes/terra-resource-id")
-  NEW_PROXY="https://${APP_PROXY}"
-  NEW_PROXY_URL="${RESOURCE_ID}.${APP_PROXY}"
-
-  # Create a systemd service to start the workbench app proxy.
-  cat << EOF > "${WORKBENCH_PROXY_AGENT_SERVICE}"
+IS_NON_GOOGLE_ACCOUNT=$(curl "https://${TERRA_SERVER/verily/terra}-user.api.verily.com/api/profile?path=non_google_account" \
```
Comment (mbookman) [inline startupscript/vertex-ai-user-managed-notebook/post-startup.sh:833]: Is there any plan to add this to the CLI?

It feels like a `wb user status` (without an email address) or `wb user profile` should let me get at some of my basic profile information, including a flag about whether it is a Google-backed account.

**Review by mbookman — COMMENTED** [review]
This looks fine, but feels like we are missing a CLI command for a user to get their own profile information? If one doesn't exist, we should file a Jira ticket to add one.

File: startupscript/vertex-ai-user-managed-notebook/post-startup.sh (hunk 1 above)
Comment (yuhuyoyo) [reply startupscript/vertex-ai-user-managed-notebook/post-startup.sh:833]: i agree.. unfortunately we don't have that expose yet. i filed https://verily.atlassian.net/browse/BENCH-2612

Comment (mbookman) [inline startupscript/vertex-ai-user-managed-notebook/post-startup.sh:833]:
```suggestion
IS_NON_GOOGLE_ACCOUNT="$(curl "https://${TERRA_SERVER/verily/terra}-user.api.verily.com/api/profile?path=non_google_account" \
```

File: startupscript/vertex-ai-user-managed-notebook/post-startup.sh [hunk 2]
```diff
@@ -830,18 +830,24 @@ fi
...
+                    -H "accept: application/json" -H "Authorization: Bearer $(gcloud auth print-access-token)" \
+                  | jq '.value')
```
Comment (mbookman) [inline startupscript/vertex-ai-user-managed-notebook/post-startup.sh:835]:
```suggestion
                  | jq '.value')"
```

File: startupscript/vertex-ai-user-managed-notebook/post-startup.sh [hunk 3]
```diff
@@ -830,18 +830,24 @@ fi
...
+readonly IS_NON_GOOGLE_ACCOUNT
+
+if [[ "${IS_NON_GOOGLE_ACCOUNT}" == "true" ]]; then
+  APP_PROXY=$(get_metadata_value "instance/attributes/terra-app-proxy")
```
Comment (mbookman) [inline startupscript/vertex-ai-user-managed-notebook/post-startup.sh:839]:
```suggestion
  APP_PROXY="$(get_metadata_value "instance/attributes/terra-app-proxy")"
```

File: startupscript/vertex-ai-user-managed-notebook/post-startup.sh [hunk 4]
```diff
@@ -830,18 +830,24 @@ fi
...
+  readonly APP_PROXY
+  TERRA_GCP_NOTEBOOK_RESOURCE_NAME="$(get_metadata_value "instance/attributes/terra-gcp-notebook-resource-name")"
+  readonly TERRA_GCP_NOTEBOOK_RESOURCE_NAME
+  if [[ -n "${APP_PROXY}" ]]; then
+    emit "Using custom Proxy Agent"
+    RESOURCE_ID=$(get_metadata_value "instance/attributes/terra-resource-id")
```
Comment (mbookman) [inline startupscript/vertex-ai-user-managed-notebook/post-startup.sh:845]:
```suggestion
    RESOURCE_ID="$(get_metadata_value "instance/attributes/terra-resource-id")"
```

File: startupscript/vertex-ai-user-managed-notebook/post-startup.sh [hunk 5]
```diff
@@ -830,18 +830,24 @@ fi
...
+    NEW_PROXY="https://${APP_PROXY}"
+    NEW_PROXY_URL="${RESOURCE_ID}.${APP_PROXY}"
```
Comment (mbookman) [inline startupscript/vertex-ai-user-managed-notebook/post-startup.sh:847]:
```suggestion
    NEW_PROXY_URL="${RESOURCE_ID}.${APP_PROXY}"
    readonly RESOURCE_ID
    readonly NEW_PROXY
    readonly NEW_PROXY_URL
```

File: startupscript/vertex-ai-user-managed-notebook/post-startup.sh [hunk 6]
```diff
@@ -855,21 +861,22 @@ Restart=on-failure
...
-  # Enable and start the startup service
-  systemctl daemon-reload
-  systemctl enable "${WORKBENCH_PROXY_AGENT_SERVICE_NAME}"
-  systemctl start "${WORKBENCH_PROXY_AGENT_SERVICE_NAME}"
-  emit "Workbench proxy Agent service started"
+    # Enable and start the startup service
+    systemctl daemon-reload
+    systemctl enable "${WORKBENCH_PROXY_AGENT_SERVICE_NAME}"
+    systemctl start "${WORKBENCH_PROXY_AGENT_SERVICE_NAME}"
+    emit "Workbench proxy Agent service started"
```
Comment (mbookman) [inline startupscript/vertex-ai-user-managed-notebook/post-startup.sh:868]:
```suggestion
    emit "Workbench Proxy Agent service started"
```

**Review by mbookman — CHANGES_REQUESTED** [review]

**Review by mbookman — APPROVED** [review]
LGTM

**Pull request merged**